### DISCUSSION TITLE
24GHz mmWave Human Static Presence Lite | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#556] [Q&A]
# 24GHz mmWave Human Static Presence Lite | Seeed Studio Wiki

24GHz mmWave Human Static Presence Lite

https://wiki.seeedstudio.com/Radar_MR24HPC1/

**Comment by taunoe** [comment]
I think there is undocumented parts in [User Manual](https://files.seeedstudio.com/wiki/mmWave-radar/MR24HPC1_User_Manual-V1.5.pdf). For example in Underlying Open function information Command word 0x06 (#defined as DETAILDIRECT) and 0x07 (DETAILSIGN). Library uses them but they are not listed in table.

**Reply by xzxcessarr** [reply]
Hello, Thanks for your feedback, we think the variable you mention is the debug interface at the development, why they are not mentioned detailed in datasheet I think it may caused by the feature or the performance of the sensor, the library we provide often contains them.

**Reply by taunoe** [reply]
When the Underlying function is turned off, every second data frame starts with a Control word 0x00 like this 53 59 00 03 00 01 06 36 54 43. What is this data? I can see it changing when there is movement. A secret again?

**Reply by xzxcessarr** [reply]
Hello, after check by our engineer, we confirm that the data frame contain the Control word 0x00 is meaningless, now our technology group is locating the reason now, after it is well I will tell you soon

**Comment by theopensourcerer** [comment]
Hi, just received a couple of these devices. If I need to flash them, I can't use j-link (don't have one) so I thought I could use UART but the PackageMake application linked to above is only for Windows. I do not have any Windows computers. I only use Linux. Is there an alternative solution to flash the device?

**Comment by NGUYENHUUCAU** [comment]
it doesn't work well when i turn on the ceiling fan

**Comment by theopensourcerer** [comment]
I'll ask again... How do update the firmware of these devices if I don't have a JLink, or a Windows computer? I only use Linux computers.

**Reply by hiteshkher** [reply]
I think not available software for Linux, so just my opinion easy way,  use friends or neighborhood windows PC

**Comment by hiteshkher** [comment]
I try MR24HPC1 frimware update from UART but somthing wrong ( I think frimware update proses start and faild ) so now not working,  as well as  MR24HPC1 before working with old frimware, so now how to upload frimware ?  orignal UART frimware proses not working

**Reply by janstuyck** [reply]
I'm having the same problem. Uploading the firmware as described over UART and with PackageMake-v1.1.1, the process bar hung at I think it was 89%. Since then no more communication.

Is this a bricked board now or is there a way to fix this?

**Reply by janstuyck** [reply]
* By "bricked" I mean broken. My question is if this device can ever work again. And if so how to accomplish this. It is connected to a Windows PC by a USB to UART converter. The converter is recognised in device manager and I can select the corresponding port in PackageMake-v1.1.1.

I was following the instructions about updating the firmware of the MR24HPC1 via UART: https://wiki.seeedstudio.com/Radar_MR24HPC1/#update-firmware-via-uart

When I first connected the MR24hpc1 boad it was sending characters that I could see in red colour in the PackageMake-v1.1.1 software. Exactly as described in the manual. I proceeded with the instructions to upload the latest firmware. This worked OK until the green progress bar stopped moving at 89%.
After waiting for half an hour I closed the PackageMake software and disconnected the USB to UART converter from the computer.

I started over again with the beginning of the instructions. This time there were no characters sent any more shown in red colour in PackageMake. When I try to upload the firmware I get one line of blue ASCII. The USB to UART converter is of course still recognised in device manager and the port and correct baud rate can be selected in PackageMake.

Is it possible to get the MR24HPC1 working again? Do I need J-Link for that? If so can I also use another USB-JTAG/TTL debugger from a different brand? Or do I need to send the MR24HPC1 back to you?
I also closed the PackageMake software and opened it again

**Comment by viralgala7** [comment]
We are using MR24HPB1, it was working fine, suddenly it started giving data in the format that doesn't match regular output format

**Reply by viralgala7** [reply]
The mismatched data is,
55 AA 03 07 00 05 65 01 00 01 01 76 55 AA 03 07 00 05 83 04 00 01 00 96 55 AA 03 01 00 2A 7B 22 70 22 3A 22 75 61 65 66 61 6D ...
IN WHICH after 55 , Length factor 2 bytes are AA 03 which does not match with frame length & 07 is not matching any command!! No data of error codes found in datasheet, pls reply whether this can be resolved or the chip is faulty??

**Comment by janstuyck** [comment]
SeeedStudio Doesn't seem to give support on its own platform?

**Comment by viralgala7** [comment]
Hello,

You are Right, Even I haven't got any response regarding the problem. I
haven't programmed the chip yet but it accidentally giving serial data
which is not as per the format , may be some error codes that we are unable
to decode, If  you find any other platform for your query resolution,
please let me know!! Do you also receive any Serial data from the chip in
any other format?

**Reply by janstuyck** [reply]
Hi.

I was answered yesterday by email: 

"
it looks like the device is not working due to a failed firmware refresh, I guess you can only reflash the firmware via J-link to fix the issue:
https://wiki.seeedstudio.com/Radar_MR24HPC1/#universal-method---use-j-link-to-burn-firmware
"

Would be better if they would answer here in the comments so other people can benefit from it..

Note that my MR24HPC1 is not outputting anything anymore through UART.

Now I am wondering if I need the expensive J-Link or if I can do with a cheaper alternative for ARM cortex chips like for instance https://www.amazon.de/-/en/DollaTek-Debugger-Emulator-Programmer-Downloader/dp/B07PQPF22L/ref=sr_1_fkmr0_1?sr=8-1-fkmr0
?
